test/testcafe/spec/IdxSessionExpired_spec.js

```diff
@@ -0,0 +1,149 @@
+import { RequestMock } from 'testcafe';
+import { checkConsoleMessages, renderWidget as rerenderWidget } from '../framework/shared';
+import IdentityPageObject from '../framework/page-objects/IdentityPageObject';
+import xhrWellKnownResponse from '../../../playground/mocks/data/oauth2/well-known-openid-configuration.json';
+import xhrInteractResponse from '../../../playground/mocks/data/oauth2/interact.json';
+import xhrIdentify from '../../../playground/mocks/data/idp/idx/identify';
+import xhrAuthenticatorRequiredPassword from '../../../playground/mocks/data/idp/idx/authenticator-verification-password';
+import xhrSessionExpired from '../../../playground/mocks/data/idp/idx/error-401-session-expired';
+
+
+const baseNetworkMocks = RequestMock()
+  .onRequestTo('http://localhost:3000/idp/idx/introspect')
+  .respond(xhrIdentify, 200)
+  .onRequestTo('http://localhost:3000/idp/idx/identify')
+  .respond(xhrAuthenticatorRequiredPassword)
+  .onRequestTo('http://localhost:3000/idp/idx/challenge/answer')
+  .respond(xhrSessionExpired, 401);
+
+const sessionExpiresDuringPasswordChallenge = baseNetworkMocks
+  .onRequestTo('http://localhost:3000/idp/idx/introspect')
```

Comment:
2nd introspect should not be necessary, what happens if this is removed?

Reply:
fixed